neo/test/coretest/test_container.py

```diff
@@ -98,10 +102,190 @@ def test__children(self):
+        self.assertEqual(2, len(seg.filter(test=neo.FilterGreaterThan(4))))
+
+    def test_filter_greater_than_equal(self):
+        '''
+            Tests FilterCondition object "FilterGreaterThanEqual".
+        '''
+
+        seg = neo.core.Segment()
+        st1 = neo.core.SpikeTrain([1, 2] * q.ms, t_stop=10)
+        st1.annotate(test=5)
+        st2 = neo.core.SpikeTrain([3, 4] * q.ms, t_stop=10)
+        st2.annotate(test=6)
+        seg.spiketrains.append(st1)
+        seg.spiketrains.append(st2)
+
+        self.assertEqual(0, len(seg.filter(test=neo.FilterGreaterThanEqual(7))))
+        self.assertEqual(1, len(seg.filter(test=neo.FilterGreaterThanEqual(6))))
+        self.assertEqual(2, len(seg.filter(test=neo.FilterGreaterThanEqual(5))))
+
+    def test_filter_is_in(self):
+        '''
+        Tests FilterCondition object "FilterIsIn".
+        '''
+
+        seg = neo.core.Segment()
+        st1 = neo.core.SpikeTrain([1, 2] * q.ms, t_stop=10)
+        st1.annotate(test=5)
+        st2 = neo.core.SpikeTrain([3, 4] * q.ms, t_stop=10)
+        st2.annotate(test=6)
+        seg.spiketrains.append(st1)
+        seg.spiketrains.append(st2)
+
+        self.assertEqual(0, len(seg.filter(test=neo.FilterIsIn([4, 7, 10]))))
+        self.assertEqual(1, len(seg.filter(test=neo.FilterIsIn([5, 7, 10]))))
+        self.assertEqual(2, len(seg.filter(test=neo.FilterIsIn([5, 6, 10]))))
+
+    def test_filter_in_range(self):
+        '''
+        Tests FilterCondition object "FilterInRange".
+        '''
+        seg = neo.core.Segment()
+        st1 = neo.core.SpikeTrain([1, 2] * q.ms, t_stop=10)
+        st1.annotate(test=5)
+        st2 = neo.core.SpikeTrain([3, 4] * q.ms, t_stop=10)
+        st2.annotate(test=6)
+        seg.spiketrains.append(st1)
+        seg.spiketrains.append(st2)
+
+        self.assertEqual(2, len(seg.filter(test=neo.FilterInRange(5, 6, False, False))))
+        self.assertEqual(1, len(seg.filter(test=neo.FilterInRange(5, 6, True, False))))
+        self.assertEqual(1, len(seg.filter(test=neo.FilterInRange(5, 6, False, True))))
+        self.assertEqual(0, len(seg.filter(test=neo.FilterInRange(5, 6, True, True))))
+
+    def test_filter_connectivity(self):
+        '''
+        Tests old functionality with new filter method.
+        '''
+        seg = neo.core.Segment()
+        st1 = neo.core.SpikeTrain([1, 2] * q.ms, t_stop=10)
+        st1.annotate(test=5)
+        st2 = neo.core.SpikeTrain([3, 4] * q.ms, t_stop=10)
+        st2.annotate(filt=6)
+        st2.annotate(name='st_num_1')
+        seg.spiketrains.append(st1)
+        seg.spiketrains.append(st2)
+
+        self.assertEqual(2, len(seg.filter({'test': neo.FilterEqual(5),
+                                            'filt': neo.FilterEqual(6)})))
+        self.assertEqual(0, len(seg.filter([{'test': neo.FilterEqual(5)},
+                                            {'filt': neo.FilterEqual(6)}])))
+        self.assertEqual(1, len(seg.filter(name='st_num_1')))
+
```

Comment:
You don't test cases yet where the annotations could be a list or similar.

Reply:
I added tests for a list and for a dictionary